server/issue_test.go

```diff
@@ -312,54 +312,61 @@ func TestRouteShareIssuePublicly(t *testing.T) {
 }
 
 func TestRouteAttachCommentToIssue(t *testing.T) {
-	api := &plugintest.API{}
-
-	api.On("LogWarn", mockAnythingOfTypeBatch("string", 13)...).Return(nil)
-
-	api.On("LogDebug", mockAnythingOfTypeBatch("string", 11)...).Return(nil)
-
-	api.On("GetPost", "error_post").Return(nil, &model.AppError{Id: "1"})
-	api.On("GetPost", "post_not_found").Return(nil, (*model.AppError)(nil))
-
-	api.On("GetPost", "0").Return(&model.Post{UserId: "0"}, (*model.AppError)(nil))
-	api.On("GetUser", "0").Return(nil, &model.AppError{Id: "1"})
-
-	api.On("GetPost", "1").Return(&model.Post{UserId: "1"}, (*model.AppError)(nil))
-	api.On("GetUser", "1").Return(&model.User{Username: "username"}, (*model.AppError)(nil))
-	api.On("CreatePost", mock.AnythingOfType("*model.Post")).Return(&model.Post{}, (*model.AppError)(nil))
-
-	api.On("PublishWebSocketEvent", "update_defaults", mock.AnythingOfType("map[string]interface {}"), mock.AnythingOfType("*model.WebsocketBroadcast"))
-
 	type requestStruct struct {
 		PostID      string `json:"post_id"`
 		InstanceID  string `json:"instance_id"`
 		CurrentTeam string `json:"current_team"`
 		IssueKey    string `json:"issueKey"`
 	}
 
-	tests := map[string]struct {
+	type testCase struct {
 		method       string
 		header       string
 		request      *requestStruct
 		expectedCode int
-	}{
+		setupMocks   func(api *plugintest.API)
+	}
+
+	successfulUserPostSetup := func(api *plugintest.API) {
+		api.On("GetPost", "1").Return(&model.Post{UserId: "1", ChannelId: "test_channel"}, (*model.AppError)(nil)).Once()
+		api.On("GetUser", "1").Return(&model.User{Username: "username"}, (*model.AppError)(nil)).Once()
+		api.On("GetChannelMember", mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(nil, nil).Once()
```

Comment:
This should probably not return nil, nil.  It can return an empty ChannelMember instead.  
```suggestion
		api.On("GetChannelMember", mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(&model.ChannelMember{}, nil).Once()
```